openstf/model/objective.py

```diff
@@ -31,19 +33,81 @@ class RegressorObjective:
         study.optimize(objective)
     """
 
+    def process_tuple(
+        self,
+        trial: optuna.trial.FrozenTrial,
+        key: str,
+        value: Tuple[Union[int, float], Union[int, float]],
+    ) -> Union[int, float]:
+        """Pick hyperparameter for tuple"""
+        tup, log = value
+        startValue, endValue = tup
+
+        error = "input values for the tuple can be float/float, float/int, int/float, int/int"
+
+        if isinstance(startValue, float) or isinstance(endValue, float):
+            return trial.suggest_float(key, float(startValue), float(endValue), log=log)
+
+        elif isinstance(startValue, int) and isinstance(endValue, int):
+            return trial.suggest_int(key, startValue, endValue)
+
+        else:
+            raise TypeError(error)
+
+    def get_trial_parameters(
+        self, model_params: dict, trial: optuna.trial.FrozenTrial
+    ) -> dict:
+        """
+        Pick hyperparameters from the hyperparameter space using optuna.
+        The dictionary can contain 2 possible types: tuple or List
+        1. a list can directly be used in the suggest_categorical
+        2. For the tuple we need a check to see what the tuple contains
+            a. two strings; it will pick one of the values based on the model used
+            b. one float with one integer or two floats;
+            will set the first value as lowerbound and the second value as upperbound and picks floats between those bounds.
+            This uses the log boolean which is defined in the tuple
+            c. two integers;
+            will set the first value as lowerbound and the second value as upperbound and picks integers between those bounds.
+            log boolean is ignored.
+
+        """
+        param = {}
+        for key, value in model_params.items():
+            # In the default parameter space we make use of tuples ((startValue: Union[int, float], endValue: Union[int, float]),log: bool)
+            if isinstance(value, tuple):
+                param[key] = self.process_tuple(trial, key, value)
+            # If the parameter is a list it means it is a categorical feature
+            elif isinstance(value, list):
+                param[key] = trial.suggest_categorical(key, value)
+
+            else:
+                raise TypeError("Possible input values: tuple, list")
+
+        return param
```

Comment:
or `params` perhaps.

Reply:
Or even better `trial_parameters` -> the `get_trial_parameters` function returns `trial_parameters`. I'm also fin with using params instead of parameters ofc.